util/resource/resource.go

```diff
@@ -0,0 +1,76 @@
+/*
```

Comment:
Should this be an exp/ internal package instead of exposing it as an API?

Reply:
This is shared with clusterctl. To use it in CRS too, I recently moved this out of clusterctl.